notificationworkerlambda/src/main/scala/com/gu/notifications/worker/utils/Cloudwatch.scala

```diff
@@ -59,6 +67,23 @@ class CloudwatchImpl(val senderMetricNs: String) extends Cloudwatch {
     }
   }
 
+  def sendLatencyMetrics(shouldPushMetricsToAws: Boolean, stage: String, platform: Option[Platform]): Pipe[IO, List[Long], Unit] = _.evalMap { deliveryTimes =>
+    IO.delay {
+      val latencyMetrics = LatencyMetrics.aggregateForCloudWatch(deliveryTimes)
+      val dimension = new Dimension().withName("platform").withValue(platform.map(_.toString).getOrElse("unknown"))
+      val requests = latencyMetrics.map { valuesAndCounts =>
+        val cloudWatchMetric: MetricDatum = latencyDatum("TokenDeliveryLatency", valuesAndCounts.uniqueValues, valuesAndCounts.orderedCounts, dimension)
+        new PutMetricDataRequest()
+          .withNamespace(s"Notifications/$stage/$senderMetricNs")
+          .withMetricData(cloudWatchMetric)
+      }
+      if (shouldPushMetricsToAws && latencyMetrics.exists(_.uniqueValues.nonEmpty)) {
+        requests.map(req => cloudwatchClient.putMetricData(req))
```

Comment:
Whilst considering the risks we speculated that there was a possibility that we could:

> Introduce invocation failures (e.g. due to a CloudWatch API call failing)

At first I wondered if all of these calls to CloudWatch should be wrapped in a `Try`, but it looks like `IO.delay` is going to help us out here:

```scala
/**
 * Suspends a synchronous side effect in `IO`.
 *
 * Any exceptions thrown by the effect will be caught and sequenced
 * into the `IO`.
 */
def delay[A](body: => A): IO[A] = {
  val nextIo = Delay(() => body)
  if (isFullStackTracing) {
    IOTracing.decorated(nextIo)
  } else {
    nextIo
  }
}
```

So I think this is safe enough.